helix-term/src/commands/typed.rs

```diff
@@ -2167,6 +2169,64 @@ fn reset_diff_change(
     Ok(())
 }
 
+fn rename_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+    ensure!(args.len() == 1, ":rename takes one argument");
+    let new_name = args.first().unwrap();
+
+    let (_, doc) = current!(cx.editor);
+    let path = doc
+        .relative_path()
```

Comment:
```suggestion
        .path()
```

using a relative path makes no sense here